### PR TITLE
ComponentName: put in separate file

### PR DESCRIPTION
**Pull request by akva2**
And introduce a compile unit.

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this please

File: opm/simulators/utils/ComponentName.cpp [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    if constexpr (Indices::enableExtbo) {
+        names_[Indices::zFractionIdx] = "ZFraction";
+    }
+
+    if constexpr (Indices::enablePolymer) {
+        names_[Indices::polymerConcentrationIdx] = "Polymer";
+    }
+
+    if constexpr (Indices::polymerMoleWeightIdx >= 0) {
+        assert(Indices::enablePolymer);
```
Comment (bska) [inline opm/simulators/utils/ComponentName.cpp:62]: Tiny nit: You need `<cassert>` for the `assert` macro.

**Review by bska — APPROVED** [review]
Looks good to me except for a very small nit.

File: opm/simulators/utils/ComponentName.cpp (hunk 1 above)
Comment (akva2) [reply opm/simulators/utils/ComponentName.cpp:64]: indeed.

**Comment by akva2** [comment]
jenkins build this please

**Comment by akva2** [comment]
merging per approval.

**Pull request merged**